pubsub/iam.js

```diff
@@ -116,114 +139,145 @@ function setSubscriptionPolicyExample (subscriptionName, callback) {
+ * Test a subcription's IAM permissions.
+ *
+ * @param {string} subscriptionName The name of the subscription.
+ * @param {function} callback The callback function.
  */
-function testSubscriptionPermissionsExample (subscriptionName, callback) {
+function testSubscriptionPermissions (subscriptionName, callback) {
+  if (!subscriptionName) {
+    return callback(new Error('"subscriptionName" is required!'));
+  }
+
   // Grab a reference to an existing subscription
   var subscription = pubsub.subscription(subscriptionName);
 
-  var tests = [
+  var permissionsToTest = [
     'pubsub.subscriptions.consume',
     'pubsub.subscriptions.update'
   ];
 
-  // Retrieve the IAM policy for the provided subscription
-  subscription.iam.testPermissions(
-    tests,
-    function (err, permissions, apiResponse) {
-      if (err) {
-        return callback(err);
-      }
-
-      console.log('Got permissions for ' + subscriptionName);
-      callback(null, permissions, apiResponse);
+  // Test the IAM policy for the specified subscription
+  subscription.iam.testPermissions(permissionsToTest, function (err, permissions) {
+    if (err) {
+      return callback(err);
     }
-  );
+
+    console.log('Tested permissions for: %s', subscriptionName);
+    return callback(null, permissions);
+  });
 }
 // [END test_subscription_permissions]
 
-exports.getTopicPolicyExample = getTopicPolicyExample;
-exports.getSubscriptionPolicyExample = getSubscriptionPolicyExample;
-exports.setTopicPolicyExample = setTopicPolicyExample;
-exports.setSubscriptionPolicyExample = setSubscriptionPolicyExample;
-exports.testTopicPermissionsExample = testTopicPermissionsExample;
-exports.testSubscriptionPermissionsExample = testSubscriptionPermissionsExample;
-
-// Run the examples
-exports.main = function (cb) {
-  var topicName = 'messageCenter2';
-  var subscriptionName = 'newMessages2';
-  async.series([
-    function (cb) {
-      createTopicExample(topicName, cb);
-    },
-    function (cb) {
-      getTopicPolicyExample(topicName, cb);
-    },
-    function (cb) {
-      testTopicPermissionsExample(topicName, cb);
-    },
-    function (cb) {
-      subscribeExample(topicName, subscriptionName, cb);
-    },
-    function (cb) {
-      getSubscriptionPolicyExample(subscriptionName, cb);
-    },
-    function (cb) {
-      testSubscriptionPermissionsExample(subscriptionName, cb);
-    },
-    function (cb) {
-      deleteSubscriptionExample(subscriptionName, cb);
-    },
-    function (cb) {
-      deleteTopicExample(topicName, cb);
+// [START usage]
+function printUsage () {
+  console.log('Usage: node iam RESOURCE COMMAND [ARGS...]');
+  console.log('\nResources:\n');
+  console.log('\ttopics');
+  console.log('\tsubscriptions');
+  console.log('\nCommands:\n');
+  console.log('\tget NAME');
+  console.log('\tset NAME');
+  console.log('\ttest NAME');
+  console.log('\nExamples:\n');
+  console.log('\tnode iam topics get my-topic');
```

Comment:
Should these 6 lines be combined into 2? (by combining the get/set/test variants into one line)